### ISSUE TITLE
Directory completion wrong when completing on user home abbreviation (~)

### ISSUE THREAD

**Issue by paugier** [xonsh/xonsh#4688]
This is similar to https://github.com/xonsh/xonsh/issues/1964, which was closed a long time ago (2017).

Tab completion of `code ~/D` leads to `code ~/D/home/pierre/D`, which is wrong.

Note that it works well with commands like `ls` or `cd`.

Also, if I set `$BASH_COMPLETIONS = []`, it works well with other programs.

```
$ xonfig                                                           
+------------------+-----------------------+
| xonsh            | 0.11.0                |
| Git SHA          | 337cf25a              |
| Commit Date      | Nov 17 15:37:41 2021  |
| Python           | 3.8.10                |
| PLY              | 3.11                  |
| have readline    | True                  |
| prompt toolkit   | 3.0.5                 |
| shell type       | prompt_toolkit        |
| history backend  | json                  |
| pygments         | 2.6.1                 |
| on posix         | True                  |
| on linux         | True                  |
| distro           | ubuntu                |
| on wsl           | False                 |
| on darwin        | False                 |
| on windows       | False                 |
| on cygwin        | False                 |
| on msys2         | False                 |
| is superuser     | False                 |
| default encoding | utf-8                 |
| xonsh encoding   | utf-8                 |
| encoding errors  | surrogateescape       |
| on jupyter       | False                 |
| jupyter kernel   | None                  |
| xontrib 1        | langenv_common        |
| xontrib 2        | pyenv                 |
| xontrib 3        | vox                   |
| xontrib 4        | voxapi                |
| RC file 1        | /home/pierre/.xonshrc |
+------------------+-----------------------+
```

**Comment by ylmrx** [comment]
The best work-around I found so far was to:

```sh
# copy upstream bash_completions to your $HOME 
$ cp /usr/share/bash-completion/bash_completion .
# edit this file in order to comment out the _expand definition
# it's near line 1000 for me, should be somewhere near for you as well
$ $BASH_COMPLETIONS=['$HOME/bash_completion']
# now you should have some (correct) bash completion for your "system installed"
# toolchain (ie. git, ssh, whatever ubuntu provides), and xonsh will handle the tilda
# expansion.
```

I'd love to come up with a less hacky solution, but this part of code is _hard_ to read.

**Comment by gforsyth** [comment]
Closed via #4667
